### DISCUSSION TITLE
Possibility to validate inputs/form via webhook

### DISCUSSION THREAD

**Discussion by botzill** [directus/directus#3004] [Archive]
I think would be good to have some validation via webhook where on save we get some response from the webhook with specific error for the field that needs proper validation. 

like

```
return {
  "field_name": "Error message"
}
```

and this way we can cover more validation use cases.

What do you think?

**Comment by rijkvanzanten** [comment]
I'm not exactly sure what you're proposing here 🤔 

Do you mean that you can have an external service control whether or not an item save succeeds?

**Comment by botzill** [comment]
Let's say we have some input with `postal_code`, now on save, webhook is called, validate that postal code and we can return back an error for that specific field `postal_code` with a custom message that is not valid.

**Reply by rijkvanzanten** [reply]
Ah I see, you can / could / should use a regular Hook for that. Webhooks that are configured through the app are run async, and won't be able to prevent an item from being saved / do validation like that.
